src/navigator-breadcrumbs.tsx

```diff
@@ -18,17 +18,46 @@
  */
 import cockpit from "cockpit";
 import React from "react";
+import { ServerIcon } from "@patternfly/react-icons";
 
 import { Button, Flex, PageBreadcrumb } from "@patternfly/react-core";
 
+function useHostname() {
+    const [hostname, setHostname] = React.useState<string | null>(null);
+
+    React.useEffect(() => {
+        const client = cockpit.dbus('org.freedesktop.hostname1');
+        const hostname1 = client.proxy('org.freedesktop.hostname1', '/org/freedesktop/hostname1');
+
+        function changed() {
+            if (hostname1.valid && typeof hostname1.Hostname === 'string') {
+                setHostname(hostname1.Hostname);
+            }
+        }
+
+        hostname1.addEventListener("changed", changed);
+        return () => {
+            hostname1.removeEventListener("changed", changed);
+            client.close();
+        };
+    }, []);
+
+    return hostname;
+}
+
 export function NavigatorBreadcrumbs({ path }: { path: string[] }) {
+    const hostname = useHostname();
+
     function navigate(n_parts: number) {
         cockpit.location.go("/", { path: encodeURIComponent(path.slice(0, n_parts).join("/")) });
     }
 
     return (
         <PageBreadcrumb stickyOnBreakpoint={{ default: "top" }}>
             <Flex spaceItems={{ default: "spaceItemsXs" }}>
+                <React.Fragment key="//hostname">
+                    <ServerIcon /> {hostname}
```

Comment:
Confusingly you want to use `HddIcon`. Also this is not clickable while we want it to be. My patch for this. https://sprunge.us/odoJRK?javascript